library/core/src/main/java/com/google/android/exoplayer2/text/SubtitleDecoderFactory.java

```diff
@@ -83,6 +85,8 @@ public SubtitleDecoder createDecoder(Format format) {
         } else if (format.sampleMimeType.equals(MimeTypes.APPLICATION_CEA708)) {
           return clazz.asSubclass(SubtitleDecoder.class).getConstructor(Integer.TYPE)
               .newInstance(format.accessibilityChannel);
+        } else if (format.sampleMimeType.equals(MimeTypes.APPLICATION_DVBSUBS) && format.initializationData != null) {
```

Comment:
Don't think you should check initData != null here. It's a bit arbitrary given you'd create the decoder if initData were non-null but empty (or full of invalid init data). It's better just to create the decoder, and have it fail later if the initData was wrong. This should never happen, since an extractor shouldn't be creating a DVB track without initData.